src/recipe/recipe.controller.spec.ts

```diff
@@ -1,18 +1,175 @@
+                preparation: 'Fetched recipe preparation',
+            };
+        }),
+        fetchAllRecipes: jest.fn(() => {
+            return [
+                {
+                    id: 0,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+                {
+                    id: 1,
+                    createdAt: Date.now(),
+                    title: 'Fetched recipe title',
+                    description: 'Fetched recipe description',
+                    ingredients: 'Fetched recipe ingredients',
+                    preparation: 'Fetched recipe preparation',
+                },
+            ];
+        }),
+        updateRecipe: jest.fn((id: number, dto: mockRecipeRequest) => {
+            return {
+                id,
+                createdAt: Date.now(),
+                title: dto.title || 'latest title',
+                description: dto.description || 'latest description',
+                ingredients: dto.ingredients || 'latest ingredients',
+                preparation: dto.preparation || 'latest preparation',
+            };
+        }),
+        deleteRecipe: jest.fn((_id: number) => {
+            return Promise<void>
+        })
+    };
+    beforeEach(async () => {
+        const module: TestingModule = await Test.createTestingModule({
+            controllers: [RecipeController],
+            providers: [
+                {
+                    provide: RecipeService,
+                    useValue: mockRecipeService,
+                },
+                PrismaService,
+            ],
+        }).compile();
+
+        prismaService = module.get<PrismaService>(PrismaService);
+        recipeService = module.get<RecipeService>(RecipeService);
+        recipeController = module.get<RecipeController>(RecipeController);
+    });
+
+    it('should be defined', () => {
+        expect(recipeController).toBeDefined();
+    });
+
+    describe('CreateRecipe', () => {
+        it('should create recipe', async () => {
+            //given
+            const dto = {
+                title: 'fries',
+                description: '>>> mcdonalds',
+                ingredients: 'potatoes',
+                preparation: 'peel and fry',
+            };
+
+            //when
+            const createdRecipe = await recipeController.createRecipe(dto);
+
+            //then
+            expect(createdRecipe).toEqual({
+                ...mockRecipe,
+            });
+        });
+    });
+    describe('FetchRecipe', () => {
+        it('should fetch recipe given id', async () => {
+            //given
+            const id = '1';
+
+            //when
+            const fetchedRecipe = await recipeController.fetchRecipe(+id);
```

Comment:
if you pass a number then you don't have to parse it here:
```suggestion
            const fetchedRecipe = await recipeController.fetchRecipe(id);
```